File: tests/SwiftUI.Tests/ColorTests.cs
using System;
using System.Reflection;
using Xunit;

namespace SwiftUI.Tests
{
    public class ColorTests : TestFixture
    {
        [Fact]
        public void AllStaticColorCallsWork ()
        {
            // In case we get an EntryPointNotFoundException 
            var exception = Record.Exception (() => {
                Type t = typeof (Color);
                var properties = t.GetProperties (BindingFlags.Public | BindingFlags.Static);
                foreach (var item in properties) {
                    var colour = item.GetValue (null);
                    Assert.NotNull (colour);
                };

            });

            Assert.Null (exception);
        }

        [Fact]
        public void ColorHSBOCallWorks ()
        {
            // In case we get an EntryPointNotFoundException 
            var exception = Record.Exception (() => {
                var colour = new Color (0.0f, 0.6f, 0.0f, 0.5f);

                Assert.NotNull (colour);
            });

            Assert.Null (exception);
        }

        [Fact]
        public void ColorColorSpaceWOCallWorks ()
        {
            // In case we get an EntryPointNotFoundException
            var exception = Record.Exception (() => {
                var colour = new Color (RGBColorSpace.DisplayP3, 0.6f, 0.5f);

                Assert.NotNull (colour);
            });

            Assert.Null (exception);
        }

        [Fact]
        public void ColorColorSpaceRGBOCallWorks ()
        {
            // In case we get an EntryPointNotFoundException
            var exception = Record.Exception (() => {
                var colour = new Color (RGBColorSpace.sRGB, 0.0f, 0.6f, 0.0f, 0.5f);

                Assert.NotNull (colour);
            });

            Assert.Null (exception);
        }
    }
}

File: tests/SwiftUI.Tests.FSharp/ColorTests.fs
module SwiftUI.Tests.FSharp.ColorTests

open System
open System.Reflection

open Xunit

open Swift.Interop

open SwiftUI

[<Fact>]
let ``AllStaticColorCallsWork`` () =
    let t = typeof<Color>;
    let properties = t.GetProperties ()
    for item in properties do
        let colour = item.GetValue (null);
        Assert.NotNull (colour);

[<Fact>]
let ``ColorHSBOCallWorks`` () =
    let colour = new Color(0.0, 0.6, 0.0, 0.5)
    Assert.NotNull(colour)

[<Fact>]
let ``ColorColorSpaceRGBOCallWorks`` () =
    let colour = new Color(RGBColorSpace.DisplayP3, 0.0, 0.6, 0.0, 0.5)
    Assert.NotNull(colour)

[<Fact>]
let ``ColorColorSpaceWOCallWorks`` () =
    let colour = new Color(RGBColorSpace.sRGB, 0.0, 0.6)
    Assert.NotNull(colour)